.github/workflows/build_and_test.yml

```
@@ -21,31 +21,49 @@ env:
 jobs:
   cancel_previous_runs:
     name: Cancel Previous Runs
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-latest
     steps:
       - uses: styfle/cancel-workflow-action@0.11.0
         with:
           access_token: ${{ secrets.GITHUB_TOKEN }}
 
   build-test:
-    runs-on: ubuntu-22.04
+    runs-on: ${{ matrix.host }}
     strategy:
       fail-fast: false
       matrix:
         include:
           - flavor_id: sidechain
             mode: sidechain
+            host: integritee-builder-sgx
+            sgx_mode: HW
           - flavor_id: offchain-worker
             mode: offchain-worker
+            host: integritee-builder-sgx
+            sgx_mode: HW
           - flavor_id: teeracle
             mode: teeracle
+            host: integritee-builder-sgx
+            sgx_mode: HW
           - flavor_id: sidechain-evm
             mode: sidechain
             additional_features: evm
+            host: integritee-builder-sgx
+            sgx_mode: HW
 
     steps:
       - uses: actions/checkout@v3
 
+      - name: Set env
+        run: |
+          if [[ ${{ matrix.sgx_mode }} == 'HW' ]]; then
+             echo "DOCKER_DEVICES=--device=/dev/sgx/enclave --device=/dev/sgx/provision" >> $GITHUB_ENV
+             echo "DOCKER_VOLUMES=-v /var/run/aesmd:/var/run/aesmd" >> $GITHUB_ENV
```

Please use long form where possible:
```suggestion
             echo "DOCKER_VOLUMES=--volume /var/run/aesmd:/var/run/aesmd" >> $GITHUB_ENV
```